src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,44 @@
+/* -------------------------------------------------------------------------- */
+/*                                   imports                                  */
+/* -------------------------------------------------------------------------- */
+import Popup from "./Popup.js";
+
+/* -------------------------------------------------------------------------- */
+/*                         PopupWithConfirmation class                        */
+/* -------------------------------------------------------------------------- */
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._popupSelector = document.querySelector(popupSelector);
+  }
+
+  setSubmitAction(action) {
+    this._submitButton = action;
+  }
+
+  _handleEscClose() {
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change it inside and only call it with `super`. The parent methods are available in any inheritor of the class.